pinot-servicemanager/build.gradle.kts

```diff
@@ -1,9 +1,41 @@
 plugins {
+  `java-library`
   id("org.hypertrace.docker-publish-plugin")
 }
 
 hypertraceDocker {
   defaultImage {
     imageName.set("pinot-servicemanager")
+    tasks.named(buildTaskName) {
+      dependsOn("copyPlugins")
+    }
   }
 }
+
+configure<JavaPluginExtension> {
+  sourceCompatibility = JavaVersion.VERSION_11
+  targetCompatibility = JavaVersion.VERSION_11
+}
+
+val plugins by configurations.creating
+
+dependencies {
+  compileOnly("org.apache.pinot:pinot-spi:0.5.0")
+  compileOnly("org.apache.pinot:pinot-avro-base:0.5.0")
+  compileOnly("org.apache.kafka:kafka-streams:5.5.1-ccs")
+  compileOnly("org.apache.kafka:kafka-clients:5.5.1-ccs")
+  implementation("org.hypertrace.core.kafkastreams.framework:kafka-streams-serdes:0.1.11") {
+    // disable the transitive dependencies and use them from pinot itself.
+    isTransitive = false
+  }
+}
+
+dependencies {
```

Comment:
Is this circularly creating a project dependency on itself? This should move to a separate module.

Reply:
Will handle this as part of the UDF PR since it's conflicted from this anyway.

Reply:
Thanks for moving to `pinot-avro-serde` as part of your PR.